### PR TITLE
Silence reentrant libevent warning except on debug builds

### PR DESCRIPTION
**Pull request by rhc54**

### CODE REVIEW

**Comment by rhc54** [comment]
@artpol84 @hppritcha @hjelmn Can one or more of you check to see if this has the desired effect? I have no way of testing it.

File: opal/mca/event/libevent2022/libevent/event.c [hunk 1]
```diff
@@ -1566,8 +1566,11 @@ event_base_loop(struct event_base *base, int flags)
 	EVBASE_ACQUIRE_LOCK(base, th_base_lock);
 
 	if (base->running_loop) {
+/*****   OMPI change   ****/
+#ifdef OPAL_DEBUG
```
Comment (jsquyres) [inline opal/mca/event/libevent2022/libevent/event.c]: @rhc54 How about using `#if OPAL_ENABLE_DEBUG`?

**Comment by artpol84** [comment]
@rhc54 @jsquyres @jladd-mlnx This indeed fixes the problem of "warn" flooding in our installation.

**Comment by jsquyres** [comment]
Please see https://github.com/rhc54/ompi/pull/5.

**Comment by artpol84** [comment]
@rhc54  @jsquyres I'm still seeing this issue in the latest master.

**Comment by artpol84** [comment]
@rhc54 @jsquyres 
It seems that since I'm using master `OPAL_ENABLE_DEBUG=1` is forced by the "Early development override"
https://github.com/open-mpi/ompi/blob/master/config/opal_configure_options.m4#L105

**Comment by rhc54** [comment]
I just realized that the libevent portion of this PR never got merged into the master for some weird reason. I'm fixing that now.

@artpol84 The behavior you are seeing is intentional - the reentrant error still exists and needs to be addressed, so we want that warning to show on developer builds. The intent here is strictly to avoid confusing users with it as (a) there is nothing they can do about it, and (b) the problem eventually works itself out after repeated cycling thru opal_progress (at least, it does that at the moment - we need to fix the problem in case timing changes and it no longer finds a way out of the reentrant loop).

**Comment by artpol84** [comment]
@rhc54 I see, but this error drives my MTT tests crazy. I have to go there and comment it out each time. Not a big deal though.

**Comment by jsquyres** [comment]
@artpol84 Should get fixed by #1168.

**Comment by rhc54** [comment]
@jsquyres Actually, it probably won't if @artpol84 is testing against a git clone instead of a tarball. So what we probably need to do is add some smarts to the pmix configure.m4 to indicate that if --disable-debug is given, then we need to pass that down to the pmix embedded code.

**Pull request closed without merging**